### PR TITLE
Documentation for kubectl plugins

### PR DESCRIPTION
**Pull request by fabianofranz**
Starting the documentation for `kubectl plugins`, a new alpha feature for the v1.8 release. 

https://github.com/kubernetes/features/issues/371

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5294)

### CODE REVIEW

**Comment by fabianofranz** [comment]
@kubernetes/sig-cli-pr-reviews PTAL, this is the start of `kubectl` plugins documentation and also a good summary of the current state of plugins support. Thanks!

**Comment by fabianofranz** [comment]
cc @monopole @smarterclayton

**Comment by steveperry-53** [comment]
@fabianofranz, I have some suggestions for the text, and I think it would be easiest if I could do an edit pass. Is the **Allow edits by maintainers** check box available in the right column? If so, could you check it. Thanks.

**Comment by fabianofranz** [comment]
@steveperry-53 it's checked now, thanks!

File: docs/tasks/extend-kubectl/kubectl-plugins.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+Second, the `plugins` directory under the user's kubeconfig dir. This means, in most cases, `~/.kube/plugins`.
+
+```shell
+# Loads plugins from both /path/to/dir1 and /path/to/dir2
+KUBECTL_PLUGINS_PATH=/path/to/dir1:/path/to/dir2 kubectl plugin -h
+```
+
+## Writing kubectl plugins
+
+Plugins can be written in any programming laguage or script that allows you to write command line commands. They can even consist in no binary at all and just rely on operating system utilities, like `echo`, `sed`, `grep`, the `kubectl` binary, and so on.
```
[the remark this answers is not in the document]
Comment (fabianofranz) [reply docs/tasks/extend-kubectl/kubectl-plugins.md]: oops :)

**Comment by steveperry-53** [comment]
@fabianofranz, See https://github.com/kubernetes/kubernetes.github.io/pull/5294/commits/daad68cce4a155e3762a293656c46d03900eb896 for suggested edits.

File: docs/tasks/extend-kubectl/kubectl-plugins.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (steveperry-53) [reply docs/tasks/extend-kubectl/kubectl-plugins.md]: Fixed in https://github.com/kubernetes/kubernetes.github.io/pull/5294/commits/daad68cce4a155e3762a293656c46d03900eb896.

**Comment by steveperry-53** [comment]
@fabianofranz, @tengqm, Does this PR seem ready to merge now?

File: docs/tasks/extend-kubectl/kubectl-plugins.md [hunk 2]
```diff
@@ -78,13 +87,14 @@ tree:                             # allows the declaration of subcommands
...
+The preceding descriptor declares the `kubectl plugin targaryen` plugin, which has one flag named `-h | --heat`.
+When the plugin is invoked, it calls the `dracarys` binary or script, which is located in the same directory as the descriptor file. The [Accessing runtime attributes](#accessing-runtime-attributes) section describes how the `dracarys` command accesses the flag value and other runtime context.
 
 ### Recommended directory structure
 
-It is recommended that each plugin has its own subdirectory in the filesystem, preferably with the same name of the plugin command. The directory must contain the `plugin.yaml` descriptor and any binary, script, asset, or other dependency it might require.
+It is recommended that each plugin has its own subdirectory in the filesystem, preferably with the same name as the plugin command. The directory must contain the `plugin.yaml` descriptor and any binary, script, asset, or other dependency it might require.
 
-So for the example described in the previous section, we could have something like:
+For example, the directory structure for the `targaryen` could look like this:
```
Comment (fabianofranz) [inline docs/tasks/extend-kubectl/kubectl-plugins.md]: I think it's missing a word after `targarien`, probably "plugin" or "example".

**Comment by fabianofranz** [comment]
One minor comment, otherwise LGTM. Thanks a lot @steveperry-53!

File: docs/tasks/extend-kubectl/kubectl-plugins.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+```
+
+The preceding descriptor declares the `kubectl plugin targaryen` plugin, which has one flag named `-h | --heat`.
+When the plugin is invoked, it calls the `dracarys` binary or script, which is located in the same directory as the descriptor file. The [Accessing runtime attributes](#accessing-runtime-attributes) section describes how the `dracarys` command accesses the flag value and other runtime context.
+
+### Recommended directory structure
+
+It is recommended that each plugin has its own subdirectory in the filesystem, preferably with the same name as the plugin command. The directory must contain the `plugin.yaml` descriptor and any binary, script, asset, or other dependency it might require.
+
+For example, the directory structure for the `targaryen` could look like this:
```
Comment (zhangxiaoyu-zidif) [inline docs/tasks/extend-kubectl/kubectl-plugins.md]: For example, the directory structure for the `targaryen` could look like this:
->
For example, the directory structure for plugin `targaryen` could look like this:
or
For example, the directory structure for plugin `targaryen`:
These would look fine. =) I think.

**Pull request merged**